src/bootstrap/common/cognite_client.py

```diff
@@ -0,0 +1,105 @@
+import logging.config
+from typing import List, Optional
+
+# TODO: PEP 484 Stub Files issue?
+from cognite.client import ClientConfig, CogniteClient
+from cognite.client.credentials import OAuthClientCredentials
+
+from ..common.base_model import Model
+
+
+class CogniteIdpConfig(Model):
+    # fields required for OIDC client-credentials authentication
+    client_name: Optional[str]
+    client_id: str
+    secret: str
+    scopes: List[str]
+    token_url: str
+
+
+class CogniteConfig(Model):
```

Comment:
Why do you need that one? Is it because of Pydantic implicit dependency? should you just inherit from both your pydantic model and built-in CogniteConfig simultaneously?

Reply:
Which " built-in CogniteConfig"? 

1. The `ClientConfig` ([link](https://github.com/cognitedata/cognite-sdk-python/blob/d95254f64015f7690e3670ae3f92a0e6560a6deb/cognite/client/config.py#L48)) from cognite-python-sdk, 

2. or from **extractor-utils** ([link](https://github.com/cognitedata/python-extractor-utils/blob/00fc351a3036adf4bfbb17ffad9fa2913eeee68a/cognite/extractorutils/configtools/elements.py#L188))?
- extractor-utils was removed as a dependency

But I wanted to keep the `cognite` yaml-config, so I ended up defining it here as Pydantic?
In `common` as it is (should be) used in `extpipes-cli` too

Reply:
Hm... Okay. I was wondering if is it a code duplication with native ClientConfig or not.